pkg/workloads/manager.go

```diff
@@ -32,31 +37,92 @@ type Manager interface {
 	// DeleteWorkload deletes a container and all associated processes/containers.
 	// It is implemented as an asynchronous operation which returns an errgroup.Group
 	DeleteWorkload(ctx context.Context, name string) (*errgroup.Group, error)
+	// DeleteWorkloads deletes the specified workloads by name.
+	// It is implemented as an asynchronous operation which returns an errgroup.Group
+	DeleteWorkloads(ctx context.Context, names []string) (*errgroup.Group, error)
 	// StopWorkload stops the named workload.
 	// It is implemented as an asynchronous operation which returns an errgroup.Group
 	StopWorkload(ctx context.Context, name string) (*errgroup.Group, error)
-	// StopAllWorkloads stops all running workloads.
+	// StopWorkloads stops the specified workloads by name.
 	// It is implemented as an asynchronous operation which returns an errgroup.Group
-	StopAllWorkloads(ctx context.Context) (*errgroup.Group, error)
+	StopWorkloads(ctx context.Context, names []string) (*errgroup.Group, error)
 	// RunWorkload runs a container in the foreground.
 	RunWorkload(ctx context.Context, runConfig *runner.RunConfig) error
 	// RunWorkloadDetached runs a container in the background.
 	RunWorkloadDetached(runConfig *runner.RunConfig) error
 	// RestartWorkload restarts a previously stopped container.
 	// It is implemented as an asynchronous operation which returns an errgroup.Group
 	RestartWorkload(ctx context.Context, name string) (*errgroup.Group, error)
+	// RestartWorkloads restarts the specified workloads by name.
+	// It is implemented as an asynchronous operation which returns an errgroup.Group
+	RestartWorkloads(ctx context.Context, names []string) (*errgroup.Group, error)
 }
 
 type defaultManager struct {
 	runtime rt.Runtime
 }
 
 // ErrContainerNotFound is returned when a container cannot be found by name.
+// ErrInvalidWorkloadName is returned when a workload name fails validation.
 var (
 	ErrContainerNotFound   = fmt.Errorf("container not found")
 	ErrContainerNotRunning = fmt.Errorf("container not running")
+	ErrInvalidWorkloadName = fmt.Errorf("invalid workload name")
+)
+
+const (
+	// AsyncOperationTimeout is the timeout for async workload operations
+	AsyncOperationTimeout = 5 * time.Minute
 )
 
+// validateWorkloadName validates workload names to prevent path traversal attacks
+// and other security issues. Workload names should only contain alphanumeric
+// characters, hyphens, underscores, and dots.
+var workloadNamePattern = regexp.MustCompile(`^[a-zA-Z0-9._-]+$`)
+
+func validateWorkloadName(name string) error {
```

Comment:
Minor stylistic point: I generally put public methods at the top of the file, and private methods after. I find it makes it easier to ignore implementation details when reading through the implementation of a struct.